matchms/SpectrumProcessor.py

```diff
@@ -268,6 +268,7 @@ def __str__(self):
     + ["clean_adduct",
        "derive_inchi_from_smiles",
        "derive_smiles_from_inchi",
```

Comment:
These are the only functional changes, the rest is restructuring of the files.

Reply:
@florian-huber I noticed I did not request a review yet, but the other PR's all build on this one. 
The main changes are moving functions to other files, to only functional change is here, changing the ProcessingReport output.